tests/callbacks/test_TimerListener.cpp

```diff
@@ -0,0 +1,272 @@
+}
+
+//! @test Check the behavior of @ref Kokkos::utils::callbacks::ParallelForTimerListener.
+TEST_F(TimerListenerTest, parallel_for)
+{
+    const auto par_for_timer_matching = std::make_shared<parallel_for_timer_t>(
+        EventRegexMatcher{.regex = std::regex("computation - level 0 - pfor")},
+        exec
+    );
+    const auto par_for_timer_no_match = std::make_shared<parallel_for_timer_t>(
+        EventRegexMatcher{.regex = std::regex("this-should-not-match")},
+        exec
+    );
+
+    Manager::register_listener(par_for_timer_matching);
+    Manager::register_listener(par_for_timer_no_match);
+
+    // TODO check which callbacks are activated at this point
+
+    MyWorkload<execution_space>{}.execute(exec);
+
+    ASSERT_GT(par_for_timer_matching->timer.duration(), Kokkos::utils::timer::seconds(0));
+
+    EXPECT_FALSE(par_for_timer_no_match->connected() || par_for_timer_no_match->closed())
+        << "The listener should not be connected or closed because the events did not match.";
+
+    Manager::unregister_listener(par_for_timer_matching.get());
+    Manager::unregister_listener(par_for_timer_no_match.get());
+}
+
+//! Listener to time regions whose name matches a regex.
+using region_timer_t = RegionTimerListener<EventRegexMatcher, execution_space>;
+
+//! @test Check traits of @ref Kokkos::utils::callbacks::RegionTimerListener.
+TEST(RegionTimerListener, traits)
+{
+    /// Check that @ref region_timer_t is a listener for @ref Kokkos::utils::callbacks::PushRegionEvent
+    /// and @ref Kokkos::utils::callbacks::PopRegionEvent.
+    static_assert(std::same_as<
+        listener_event_type_list_t<region_timer_t>,
+        Kokkos::Impl::type_list<PushRegionEvent, PopRegionEvent>
+    >);
+
+    static_assert(Listener<region_timer_t>);
+}
+
+//! @test Check the behavior of @ref Kokkos::utils::callbacks::RegionTimerListener.
+TEST_F(TimerListenerTest, region)
+{
+    const auto par_for_timer = std::make_shared<parallel_for_timer_t>(
+        EventRegexMatcher{.regex = std::regex("computation - level 0 - pfor")},
+        exec
+    );
+
+    const auto region_timer = std::make_shared<region_timer_t>(
+        EventRegexMatcher(std::regex("computation - level 0"))
+    );
+
+    Manager::register_listener(par_for_timer);
+    Manager::register_listener(region_timer);
+
+    MyWorkload<execution_space>{}.execute(exec);
+
+    ASSERT_GT(par_for_timer->timer.duration(), Kokkos::utils::timer::milliseconds{0.});
+    ASSERT_LT(par_for_timer->timer.duration(), region_timer->timer.duration());
+
+    Manager::unregister_listener(par_for_timer.get());
+    Manager::unregister_listener(region_timer.get());
+}
+
+//! Mesure the time for launch + time for kernel.
+template <Kokkos::utils::concepts::ExecutionSpace Exec>
+struct EnqueuedEventTimerWithLatency
```

Comment:
OK. This is nice indeed! We could also move it to its own PR. But it's fine to keep it here too.

Reply:
Still in this PR for now.